source/server/active_tcp_listener.cc

```diff
@@ -284,6 +284,10 @@ void ActiveTcpListener::resumeListening() {
 
 void ActiveTcpListener::newConnection(Network::ConnectionSocketPtr&& socket,
                                       std::unique_ptr<StreamInfo::StreamInfo> stream_info) {
+  // populate the SNI into stream info, the network filters needn't do that again.
+  if (!socket->requestedServerName().empty()) {
```

Comment:
Can this be done unconditionally? I would expect that calling `setRequestedServerName()` with an empty string would have no effect.

Reply:
It has a constructor even for empty string https://github.com/envoyproxy/envoy/blob/main/source/common/stream_info/stream_info_impl.h#L237. Or we can move the check inside the `setRequestedServerName()`, but that will block the case when want to set the server name back to empty. Although I'm not sure we have any case to set the server name back to empty, but it still feels bad for an interface.

Reply:
I think it is functionally the same without this check; setting it to empty is the same as not setting it. Please remove the conditional to simplify the code. Same in `unlink()`.

Reply:
got it, let me remove it.